proto/dydxprotocol/vault/query.proto

```diff
@@ -22,3 +28,25 @@ message QueryParamsRequest {}
 message QueryParamsResponse {
   Params params = 1 [ (gogoproto.nullable) = false ];
 }
+
+// QueryVaultRequest is a request type for the Vault RPC method.
+message QueryVaultRequest {
+  VaultType type = 1;
+  uint32 number = 2;
+}
+
+// QueryVaultResponse is a response type for the Vault RPC method.
+message QueryVaultResponse {
+  VaultId vault_id = 1 [ (gogoproto.nullable) = false ];;
+  dydxprotocol.subaccounts.SubaccountId subaccount_id = 2 [ (gogoproto.nullable) = false ];;
+  uint64 equity = 3;
+  uint64 inventory = 4;
+  uint64 total_shares = 5;
+  repeated OwnerShares all_owner_shares = 6 [ (gogoproto.nullable) = false ];;
```

Comment:
nit: Might be better to have a separate paginated query for this, considering there could be a lot of owners.

Reply:
makes sense. created a ticket to add a separate query for owner shares. let me remove owner shares for now